starboard/shared/win32/posix_emu/include/arpa/inet.h

```diff
@@ -17,12 +17,10 @@
 
 #include <stdint.h>
 #include <stdlib.h>
+// clang-format off
 #include <winsock2.h>
 #include <ws2tcpip.h>
-
-#undef NO_ERROR  // http://b/302733082#comment15
-//  TODO: b/324981660 undefine the caller of this GetCurrentTime in
-//  <winsock2.h>
-#undef GetCurrentTime
+#include <_remove_problematic_windows_macros.h>
```

Comment:
Is the leading underscore useful here ? Perhaps just `#include "remove_problematic_windows_macros.h"` ?

Reply:
If anything, shouldn't we add something that would make clang-format automatically put it at the end of a list of includes?

Reply:
Given that this is a local header, this should use quote marks `"`, rather than angle brackets `<`. If you do this, clang-format will move it below other headers.

If you really want to make sure it's last and stands out, `remove_problematic_windows_marcos.inc` would work as well. ( this is sort of abusing the non-self-contained header rule: https://google.github.io/styleguide/cppguide.html#Self_contained_Headers )

Reply:
Sample:
```
~$ cat foo.cpp 
#include "mystuff.inc"
#include "elsethis.h"
#include <something.h>

~$ clang-format foo.cpp 
#include <something.h>

#include "elsethis.h"
#include "mystuff.inc"
```

Reply:
In principle we can also push this into a separately handled include category with clang-format config: https://clang.llvm.org/docs/ClangFormatStyleOptions.html#includecategories